File: Solutions/JavaScript/src/LIFOCache.js
const Cache = require('./Cache');

class LIFOCache extends Cache {
    #capacity;
    constructor(capacity) {
        super();
        this.#capacity = capacity;
    }

    add(key, value) {
        throw new Error("Method not implemented.");
    }

    get(key) {
        throw new Error("Method not implemented.");
    }

    delete(key) {
        throw new Error("Method not implemented.");
    }

    get size() {
        throw new Error("Method not implemented.");
    }

    get capacity() {
        return this.#capacity;
    }
}

module.exports = LIFOCache;

File: Solutions/Java/app/src/main/java/LIFOCache.java

public class LIFOCache<K, V> extends Cache<K, V> {
    private final int capacity;

    public LIFOCache(int capacity) {
        this.capacity = capacity;
    }

    @Override
    public void add(K key, V value) {
        throw new UnsupportedOperationException("Not implemented yet");
    }

    @Override
    public V get(K key) {
        throw new UnsupportedOperationException("Not implemented yet");
    }

    @Override
    public void delete(K key) {
        throw new UnsupportedOperationException("Not implemented yet");
    }

    @Override
    public int size() {
        throw new UnsupportedOperationException("Not implemented yet");
    }

    @Override
    public int capacity() {
        return this.capacity;
    }
}